src/test-project/src/test/java/com/redhat/devtools/intellij/commonuitest/fixtures/test/dialogs/settings/SettingsDialogTest.java

```diff
@@ -0,0 +1,115 @@
+import static com.intellij.remoterobot.utils.RepeatUtilsKt.waitFor;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * SettingsDialog test
+ *
+ * @author zcervink@redhat.com
+ */
+class SettingsDialogTest extends LibraryTestBase {
+    private static FlatWelcomeFrame flatWelcomeFrame;
+    private static SettingsDialog settingsDialog;
+
+    @BeforeAll
+    public static void openSettingsDialog() {
+        flatWelcomeFrame = remoteRobot.find(FlatWelcomeFrame.class, Duration.ofSeconds(10));
+        flatWelcomeFrame.openSettingsDialog();
+        settingsDialog = remoteRobot.find(SettingsDialog.class, Duration.ofSeconds(5));
+    }
+
+    @AfterAll
+    public static void closeSettingsDialog() {
+        settingsDialog.cancel();
+    }
+
+    @Test
+    public void navigateToTest() {
+        settingsDialog.navigateTo("Appearance & Behavior", "Notifications");
+        try {
+            waitFor(Duration.ofSeconds(10), Duration.ofMillis(250), "The 'Notifications' settings page is not available.", () -> isSettingsPageLoaded("Notifications"));
+        } catch (WaitForConditionTimeoutException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+            fail("The 'Settings' dialog should display the 'Notifications' page.");
+        }
+
+        settingsDialog.navigateTo("Keymap");
+        try {
+            waitFor(Duration.ofSeconds(10), Duration.ofMillis(250), "The 'Keymap' settings page is not available.", () -> isSettingsPageLoaded("Keymap"));
+        } catch (WaitForConditionTimeoutException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+            fail("The 'Settings' dialog should display the 'Keymap' page.");
+        }
+    }
+
+    private boolean isSettingsPageLoaded(String label) {
+        ComponentFixture cf = remoteRobot.find(ComponentFixture.class, byXpath("//div[@class='Breadcrumbs']"));
+        return cf.hasText(label);
+    }
+
+    @Test
+    public void settingsTreeTest() {
+        JTreeFixture settingsTree = settingsDialog.settingsTree();
+        assertTrue(settingsTree.hasText("Appearance & Behavior"), "The Settings tree does not contain the 'Appearance & Behavior' item.");
+    }
+
+    @Test
+    public void okTest() {
+        settingsDialog.ok();
+
+        try {
+            remoteRobot.find(SettingsDialog.class, Duration.ofSeconds(5));
+        } catch (WaitForConditionTimeoutException e) {
+            flatWelcomeFrame.openSettingsDialog();
+            updateSettingsDialogFixture();
+            return;
+        }
+
+        fail("The 'Settings' dialog should be closed but is not.");
+    }
+
+    @Test
+    public void cancelTest() {
+        settingsDialog.cancel();
+
+        try {
+            remoteRobot.find(SettingsDialog.class, Duration.ofSeconds(5));
+        } catch (WaitForConditionTimeoutException e) {
+            flatWelcomeFrame.openSettingsDialog();
```

Comment:
If this is some refreshment of the settingsDialog before next test, it should be part of after method.

Reply:
Refactored and moved to afterclass.